messaging-client-databus/src/main/java/com/inmobi/databus/partition/CollectorReader.java

```diff
@@ -269,25 +269,11 @@ public MessageCheckpoint buildStartPartitionCheckpoints() {
   @Override
   public synchronized Long getReaderBackLog() throws IOException {
     Long collectorStreamPendingSize = 0L, localStreamPendingSize = 0L;
-    int retryCount = 0;
-    int maxRetryThreshold = 10;
-    while (cReader.getCurrentFile() == null && retryCount < maxRetryThreshold) {
-      try {
-        Thread.sleep(1000);
-      } catch (InterruptedException e) {
-        LOG.info("Sleep Interrupted while waiting for collector reader initialization. Ignoring the interrupt");
-      }
-      retryCount++;
-    }
-    //get collector reader remaining size
-    if (cReader.getCurrentFile() != null) {
-      collectorStreamPendingSize += cReader.getPendingSize(cReader.getCurrentFile());
-      LOG.info("Pending Size of uncompressed data inside collector output location starting from collector file: " +
-              cReader.getCurrentFile() + " is " + collectorStreamPendingSize);
-    }
-    //get local reader remaining size
-    retryCount = 0;
-    while (lReader.getCurrentFile() == null && retryCount < maxRetryThreshold) {
+    int retryCount = 0, maxRetryThreshold = 10;
+    /**
+     * waiting to make sure reader is initialised
+     */
+    while (reader == null && retryCount < maxRetryThreshold) {
```

Comment:
Should we say the client that reader is not yet initialized and throw an exception instead of retries?

@satish-mittal Whats the expectation for getPendingSize() if consumer is not initialized ?